### PR TITLE
Add ruby 3 compatibility

### PR DESCRIPTION
**Pull request by IvanTakarlikov-st**
This PR also includes changes from this [PR](https://github.com/sensortower/mongoid/pull/2) that adding github actions to run tests. 

Rest of changes in this PR, is basically cherry-picks of 2 PRs with rebase changes
- https://github.com/mongodb/mongoid/pull/5015 
- https://github.com/mongodb/mongoid/pull/5009

### CODE REVIEW

**Comment by IvanTakarlikov-st** [comment]
Green PR with ruby 2.7 and this version of mongoid - https://github.com/sensortower/overseer/pull/33817

**Review by sensorsasha — APPROVED** [review]
Cool. Looks safe and very straightforward to me.

File: lib/mongoid/version.rb [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # encoding: utf-8
 
 module Mongoid
-  VERSION = "7.0.13"
+  VERSION = "7.0.13r3"
```
Comment (ST-eugenekrivdyuk) [inline lib/mongoid/version.rb:4]: 👍

**Review by ST-eugenekrivdyuk — APPROVED** [review]
Can you please find or create issue collecting information about what's needed to upgrade to 7.1.x? It would be great to link it when updating overseer and adblock-web Gemfiles with this version.

**Pull request merged**